boa/src/object/internal_methods.rs

```diff
@@ -902,15 +902,23 @@ impl GcObject {
         // 4. For each element key of ownKeys, do
         for key in own_keys {
             // a. If Type(key) is String, then
-            if let PropertyKey::String(key_str) = &key {
+            let key_str = if let PropertyKey::Index(index) = &key {
+                Some(index.to_string().into())
+            } else if let PropertyKey::String(key_str) = &key {
+                Some(key_str.clone())
+            } else {
+                None
+            };
```

Comment:
```suggestion
            let key_str = if let PropertyKey::Index(index) = &key {
                Some(index.to_string().into())
            } else if let PropertyKey::String(key_str) = &key {
                Some(key_str.clone())
            } else {
                None
            };
```
I think a `match` here would make it more readable :)

Reply:
True, looks way better.